### PR TITLE
Improve client 1 to implement 1

### LINKED ISSUE #19: There should be a WebClient instead of a GitlabClient
**Describe the feature or change you'd like**
Currently the only client which has access to the scraper endpoint is the GitlabClient. Using this client for other use cases is not sufficient. Therefore a general WebClient should be implement which can handle all request to the scraper endpoint.

### LINKED ISSUE #20: There should be a URL checker for every client
**Describe the feature or change you'd like**
Currently, a client cannot process a URL string that does not contain a protocol. This should be made possible by adding a URL checker.

**Additional context**
Valid: https://foo.bar
Invalid: foo.bar

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no WebClient #19
- Currently there is no URL checker for clients. #20

**What is the new behavior (if this is a feature change)?**
- Now there is a WebClient. Closes #19
- Now there is a URL checker for clients. Closes #20

**Other information**:
- Ref: 🐍

### CODE REVIEW

**Review by Aichnerc — APPROVED** [review]
Looks good to me. 🐋

File: src/index.ts [hunk 1]
```diff
@@ -32,7 +36,18 @@ interface IClient {}
...
-  constructor(ep: IEndpoint) {}
+  constructor(ep: IEndpoint) {
+    /*
+     * When no protocol is defined, http will be appended. Therefore https
+     * should always be included for performance.
+     */
+    ep.url = ((url: string) => {
+      if (!/^(?:f|ht)tps?\:\/\//.test(url)) {
+        url = "http://" + url;
+      }
```
Comment (pinterid) [inline src/index.ts:47]: Add newline.

**Review by pinterid — CHANGES_REQUESTED** [review]
Looks good after a change.

**Review by pinterid — APPROVED** [review]
Nice.

**Pull request merged**